test/ir_Mitsubishi_test.cpp

```diff
@@ -689,11 +689,283 @@ TEST(TestMitsubishiACClass, MessageConstuction) {
+  418, 420,  416, 396,  440, 398,  440, 424,  412, 426,  420, 418,  420, 392,
+  444, 394,  444, 422,  416, 422,  414, 398,  440, 426,  422, 388,  448, 416,
+  420, 418,  418, 422,  416, 422,  414, 424,  414, 398,  438, 426,  422, 418,
+  420, 390,  446, 418,  418, 420,  416, 396,  440, 424,  412, 426,  412, 400,
+  446, 418,  420, 420,  418, 420,  416, 422,  414, 422,  414, 424,  412, 426,
+  422, 390,  446, 1232,  442, 1240,  446, 394,  444, 394,  442, 1238,  446,
+  1234,  440, 1240,  444};  // UNKNOWN F6FDB82B
+
+  irsend.sendRaw(rawData, 583, 33);
+  irsend.makeDecodeResult();
+
+  ASSERT_TRUE(irrecv.decode(&irsend.capture));
+  EXPECT_EQ(MITSUBISHI_AC, irsend.capture.decode_type);
+  uint8_t expected[18] = {0x23, 0xCB, 0x26, 0x01, 0x00, 0x00, 0x18, 0x0A, 0x36,
+                          0x79, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                          0xE6};
+  for (int i=0; i < 18; i++) {
+    EXPECT_EQ(expected[i], irsend.capture.state[i]);
+  }
+}
+
+// Tests for decodeMitsubishiAC() when the repeat mark is wrong.
+TEST(TestDecodeMitsubishiAC, DecodeRealExampleRepeatMarkError) {
+  IRsendTest irsend(4);
+  IRrecv irrecv(4);
+  irsend.begin();
+
+  irsend.reset();
+  // Mitsubishi AC "Power On, 16C, low fan, vane auto move".
+uint16_t rawData[583] = {3476, 1708,  416, 1264,  420, 1260,  414, 400,  448,
+  390,  446, 392,  444, 1236,  440, 400,  446, 392,  446, 1234,  440, 1266,
+  418, 396,  442, 1264,  420, 394,  444, 394,  442, 1264,  422, 1260,  414,
+  398,  440, 1266,  418, 1264,  420, 392,  446, 392,  444, 1264,  422, 392,
+  446, 392,  446, 1260,  414, 400,  448, 390,  446, 392,  444, 394,  442,
+  396,  442, 398,  440, 424,  412, 400,  448, 390,  446, 392,  446, 392,
+  444, 394,  442, 396,  442, 396,  440, 398,  438, 400,  448, 390,  446, 392,
+  446, 392,  444, 396,  442, 396,  440, 398,  440, 400,  438, 400,  448, 390,
+  446, 392,  444, 1236,  440, 1266,  418, 394,  442, 396,  440, 398,  438,
+  402,  446, 1232,  444, 396,  440, 1268,  418, 394,  442, 396,  440, 398,
+  440, 400,  448, 390,  448, 1232,  442, 1266,  420, 394,  444, 1264,  422,
+  1260, 416, 396,  440, 398,  450, 1230,  444, 396,  442, 398,  440, 1266,
+  418, 1264, 422, 1258,  416, 1266,  418, 394,  442, 396,  440, 398,  440,
+  398,  438, 400,  446, 392,  446, 392,  446, 392,  444, 396,  442, 396,
+  440,  398, 438, 398,  438, 400,  448, 392,  446, 392,  444, 394,  444,
+  396,  442, 396,  440, 398,  438, 400,  448, 390,  448, 392,  444, 394,
+  444, 394,  442,  396,  442, 396,  440, 398,  438, 400,  448, 390,  446,
+  392,  446, 392,  444, 394,  442, 396,  442, 396,  440, 398,  438, 400,
+  448, 390,  446, 392,  444,  394,  444, 394,  442, 396,  440, 398,  440,
+  398,  438, 400,  448, 390,  446, 392,  444, 394,  444, 394,  442, 396,
+  440, 398,  438, 400,  438, 400,  448, 392,  446, 392,  444, 394,  442, 396,
+  442, 396,  440, 398,  438, 1240,  444, 1264,  422, 390,  446, 392,  446,
+// Repeat mark (1111) wrong:
```

Comment:
Maybe we should increase the tolerance value so 1111 matches.

Reply:
This is intentionally wrong. Here I test what happen when my decoder failed to find the repeat mark&space.